redis/pool_test.go

```diff
@@ -48,21 +48,46 @@ func (c *poolTestConn) Close() error {
 func (c *poolTestConn) Err() error { return c.err }
 
 func (c *poolTestConn) Do(commandName string, args ...interface{}) (interface{}, error) {
+	return c.do(c.Conn.Do, commandName, args...)
+}
+
+func (c *poolTestConn) DoContext(ctx context.Context, commandName string, args ...interface{}) (interface{}, error) {
+	cwt, ok := c.Conn.(redis.ConnWithContext)
+	if !ok {
+		return nil, errors.New("redis: connection does not support ConnWithContext")
+	}
+	return c.do(
+		func(c string, a ...interface{}) (interface{}, error) { return cwt.DoContext(ctx, c, a...) },
+		commandName, args)
```

Comment:
Could we expand this out to make it easier to read, assumes previous change too.
```suggestion
	return c.do(
		func(command string, args ...interface{}) (interface{}, error) {
			return cwc.DoContext(ctx, command, args...)
		},
		commandName, args
	)
```

Reply:
sure, no problem.